### PR TITLE
Issue/363/issue template

### PR DESCRIPTION
**Pull request by dougbrn**
## Change Description
Adds the three issue templates created for the rail-ppt, and adds documentation on issue and PR templating. Addresses #363  

I chose to add the issue templates to the PPT as part of this PR, which is possibly outside the scope of #363. If it's preferable to just give users the documentation on how to generate these instead of creating some defaults, I'm more than happy to adjust this to do that.

## Checklist

- [x] This PR is meant for the `lincc-frameworks/python-project-template` repo and not a downstream one instead.
- [x] This change is linked to an open issue
- [x] This change includes integration testing, or is small enough to be covered by existing tests

### CODE REVIEW

File: docs/practices/issue_pr_templating.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Issue templates are highly customizable, and it's often appropriate to have
+multiple available to users to promote better organization of common issue
+types. By default, the python-project-template provides three base templates
+
+* **general-issue-template**: A general issue template for issues that don't fit into other categories.
+* **bug-report-template**: A template specific to identifying bugs within the package.
+* **feature-request-template**: A template for requesting new features of the package.
+
+A user will be prompted to choose between these templates any time they attempt
+to create a new issue in the repository. You may add to new templates to this
```
Comment (drewoldag) [inline docs/practices/issue_pr_templating.rst:26]:
```suggestion
to create a new issue in the repository. You may add new templates to this 
```

**Comment by drewoldag** [comment]
This looks pretty good, were you going to include a PR template as well?

**Comment by dougbrn** [comment]
> This looks pretty good, were you going to include a PR template as well?

Thanks @drewoldag ! A lot of this is just assuming the use of the existing PR template: https://github.com/lincc-frameworks/python-project-template/blob/main/python-project-template/.github/pull_request_template.md

I know that this one isn't quite finalized though, unsure if we wanted to try tuning this up or not. Anecdote from TAPE experience is that it's a bit overwhelming to use for me at times.

**Comment by drewoldag** [comment]
First of all 🤦,  I forgot taht we already include the PR template. 
Second, agreed it can be overwhelming, but this isn't the PR to address that :)

**Review by drewoldag — APPROVED** [review]
Looks good to me, thanks @dougbrn!

**Review by delucchi-cmu — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**